core/src/main/java/com/adobe/aem/guides/wknd/core/services/PageSearchServiceImpl.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Created on 09-08-2023 21:34 by jjedlins
+ *
+ * Copyright (c) 2001-2023 Unity S.A.
+ * ul. Strzegomska 2-4, 53-611 Wrocław, Poland
+ * Wszelkie prawa zastrzeżone
+ *
+ * Niniejsze oprogramowanie jest własnością Unity S.A.
+ * Wykorzystanie niniejszego oprogramowania jest możliwe tylko na podstawie
+ * i w zgodzie z warunkami umowy licencyjnej zawartej z Unity S.A.
+ */
+
+package com.adobe.aem.guides.wknd.core.services;
+
+import java.awt.*;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.jcr.query.Query;
+
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.jackrabbit.webdav.property.ResourceType;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.metatype.annotations.Designate;
+
+import com.day.cq.wcm.api.Page;
+
+
+@Component(service = PageSearchService.class, immediate = true)
+@Designate(ocd = PageSearchConfig.class)
+public class PageSearchServiceImpl implements PageSearchService {
+
+
+	private String rootPath;
+
+	@Activate
+	public void activate(PageSearchConfig pageSearchConfig) {
+		this.rootPath = pageSearchConfig.getRootPath();
+	}
+
+	@Override
+	public String getRootPath() {
```

Comment:
I think this method is redundant, root path can be used directly in executeQuery method.